site/app/libraries/Logger.php

```diff
@@ -198,4 +198,33 @@ public static function logAccess($user_id, $token, $action) {
         $log_message = implode(" | ", $log_message)."\n";
         @file_put_contents(FileUtils::joinPaths(static::$log_path, 'access', "{$filename}.log"), $log_message, FILE_APPEND | LOCK_EX);
     }
+
+
+    /**
+     * This logs the grading activity of any graders when they
+     * 1. Open the student's page to grade
+     * 2. Opening a component
+     * 3. Saving a component
+     * The log is in the format of
+     * Timestamp | Gradeable_id | Grader ID | Student ID | Component_ID (except for case 1) | Action | User Agent
```

Comment:
we should just have a default value so every row as the same number of columns (like id=-1)